### PR TITLE
Player movement, camera, and respawn code

### LINKED ISSUE #15: Demo use deprecated Godot Physics
Demo use deprecated Godot Physics Engine instead new Bullet Physics

### LINKED ISSUE #31: Camera clips into objects
If stand next to some object you can rotate camera into object. It should instead be pushed out.

### LINKED ISSUE #32: Falling through world
Easy to jump over railings and fall forever. Additionally, if you approach orange blocks from where red robot is, they will cause you to have vertical momentum, basically assuring you will fall over rail and into world.

### PR DESCRIPTION
**Pull request by aaronfranke**
* Adjust some of the node names to be PascalCase. This is still incomplete, will convert more later.

* Adjust player movement code to avoid redundant and confusing negatives.

* Make the camera use a [SpringArm](https://docs.godotengine.org/en/latest/classes/class_springarm.html) node, fixes #31

* Since SpringArm requires Bullet, switch to Bullet physics, fixes #15 

* Add a system for respawning the player if they fall off the map, fixes #32

### CODE REVIEW

File: player/player.gd [hunk 1]
```diff
@@ -34,12 +42,12 @@ func _ready():
...
 
 
-func rotate_camera(move):
-	$camera_base.rotate_y(-move.x)
-	$camera_base.orthonormalize() # After relative transforms, camera needs to be renormalized.
-	camera_x_rot += move.y
-	camera_x_rot = clamp(camera_x_rot, deg2rad(CAMERA_X_ROT_MIN), deg2rad(CAMERA_X_ROT_MAX))
-	$camera_base/camera_rot.rotation.x = camera_x_rot
+func _process(_delta):
+	if transform.origin.y < -17:
```
Comment (Calinou) [inline player/player.gd]: You may want to add a comment here to explain what this block does in layman's terms :slightly_smiling_face:

Comment (aaronfranke) [reply player/player.gd]: Added comments, but I don't think it's worth making constants in this case.
